### PR TITLE
[finishes #187354254] permissions and role implementation

### PR DESCRIPTION
**Pull request by JeanIrad**
## Purpose

implementing the functionalities where the Admin should be able to assign various users to various roles,
and each role should have permissions dedicated to it. 

## Changes Made

created a permission table and other permission controllers

## Testing Instructions

by providing the right parameters as described in the readme file, one can have a sound test this PR

## Related Issues

Not yet

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

- [x] Code follows the project's coding standards
- [x] Changes are covered by tests
- [x] Documentation is updated (if applicable)
- [x] All tests pass

### CODE REVIEW

**Review by Favor-star — COMMENTED** [review]
Won't you create a modal for it?! 
Definitely it's a work in a progress.

**Comment by JeanIrad** [comment]
> Won't you create a modal for it?! Definitely it's a work in a progress.

which model?

File: src/database/models/role.ts [hunk 1]
```diff
@@ -20,23 +22,30 @@ class Role extends Model<RoleAttributes, RoleCreationAttributes> implements Role
...
     id: {
-      type: DataTypes.UUID,
-      defaultValue: UUIDV4,
+      type: DataTypes.INTEGER,
       primaryKey: true,
+      autoIncrement: true,
     },
     name: {
-      type: DataTypes.STRING,
+      type: DataTypes.ENUM('admin', 'buyer', 'guest', 'vendor'),
```
Comment (amin-leon) [inline src/database/models/role.ts:30]: seller or vendor

**Review by amin-leon — APPROVED** [review]
do we have role called vendor? we are using seller?

File: src/database/models/role.ts (hunk 1 above)
Comment (JeanIrad) [reply src/database/models/role.ts:30]: vendor or seller are all the same

**Review by P-Rwirangira — APPROVED** [review]

**Review by P-Rwirangira — APPROVED** [review]

**Review by niyontwali — APPROVED** [review]

**Pull request merged**